From 7017db5123fbbccca0dde37e7a2f45172d50e56e Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Wed, 23 Dec 2020 21:18:35 +0100
Subject: [PATCH 52/96] drm/meson: Describe the HDMI PHY frequency limits of
 Meson8/8b/8m2

The maximum HDMI PLL frequency used by the vendor kernel is 2.976GHz.
For Meson8 and Meson8b (both "HDMI 1.4 4k" capable) the maximum HDMI PHY
frequency therefore is 2.976GHz. This makes sure we don't expose any
HDMI 2.0 modes.
Meson8b only supports up to 1080p according to it's datasheet. Limit the
Meson8b SoC's to 1.65GHz (similar to what's already there for GXL S805X
and S805Y).

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 drivers/gpu/drm/meson/meson_drv.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/meson/meson_drv.c b/drivers/gpu/drm/meson/meson_drv.c
index ffc5eb588..bbb8b8b50 100644
--- a/drivers/gpu/drm/meson/meson_drv.c
+++ b/drivers/gpu/drm/meson/meson_drv.c
@@ -200,13 +200,29 @@ struct meson_drm_soc_attr {
 };
 
 static const struct meson_drm_soc_attr meson_drm_soc_attrs[] = {
-	/* S805X/S805Y HDMI PLL won't lock for HDMI PHY freq > 1,65GHz */
+	/* The maximum frequency of HDMI PHY on Meson8 and Meson8m2 is ~3GHz */
+	{
+		.limits = {
+			.max_hdmi_phy_freq = 2976000,
+		},
+		.attrs = (const struct soc_device_attribute []) {
+			{ .soc_id = "Meson8 (S802)", },
+			{ .soc_id = "Meson8m2 (S812)", },
+			{ /* sentinel */ },
+		}
+	},
+	/*
+	 * GXL S805X/S805Y HDMI PLL won't lock for HDMI PHY freq > 1,65GHz.
+	 * Meson8b (S805) only supports "1200p@60 max resolution" according to
+	 * the public datasheet.
+	 */
 	{
 		.limits = {
 			.max_hdmi_phy_freq = 1650000,
 		},
 		.attrs = (const struct soc_device_attribute []) {
 			{ .soc_id = "GXL (S805*)", },
+			{ .soc_id = "Meson8b (S805)", },
 			{ /* sentinel */ }
 		}
 	},
-- 
2.45.1

